### DISCUSSION TITLE
Lock wait timeout exceeded; try restarting transaction

### DISCUSSION THREAD

**Discussion by astesin** [mydumper/mydumper#1037] [Q&A]
I'm restoring **mydumper**-obtained _(worked without a hitch)_ backup to the MariaDB 10.6.12 server. Backup was taken with options (among others):

-  -t                     6                     \ 
-  --less-locking                         \
-  -Y -G -E -R                              \
-  --tz-utc                                   \
-  -r                     5000                \
-  --order-by-primary --hex-blob

The target server was given 16GB RAM and 8 CPUs, **myloader** started at 14:50 now is 19:32
On the target server. mariadbd processes all have RSS of 9593M (innodb_buffer_pool_size = 8G)
On the target servers, CPUs are almost idle (why?) and **iostat -d** demonstrates some moderate tps rate.
The problem is:

grep ERROR myloader.log

throws multiple results starting at 16:48:47 which all are
Thread 2: Error restoring: Lock wait timeout exceeded; try restarting transaction
Error occurs between lines: 7 and 4664 on file mydb.mytable.00004.sql: Lock wait timeout exceeded; try restarting transaction
Thread 2: Error restoring: Lock wait timeout exceeded; try restarting transaction
Error occurs between lines: 4665 and 8751 on file mydb.mytable.00004.sql: Lock wait timeout exceeded; try restarting transaction
...and so on.

Does it mean that myloader have failed already, restore process failed, and I have nothing to wait for anymore?

Thanks in advance! Warmest regards, Andreas

UPD. 20:46:33 [INFO] - Thread 6: Data import ended
...
2023-02-15 20:48:11 [INFO] - Starting table checksum verification

**Comment by davidducos** [comment]
Hi @astesin,
Yes, seems that myloader have failed if the second try also failed. 
You can check the files, but it might be possible that this is a database issue, not myloader, as you might be exceeding the db resources. 
BTW, to confirm if everything is ok, you should enable the checksum verification at mydumper.

**Reply by astesin** [reply]
I gave the target server as much resources as I could - 8xCPU, 16GB RAM, and it wasn't looking at all busy while restoring.
Also I didn't disable table checksum verification, as you can see above _(I guess it is the default setting now?),_ now waiting for checksum verification to complete with impatience.

**Reply by astesin** [reply]
I've added -M option into **mydumper** launcher script for future.

The **myloader** log record 2023-02-15 20:48:11 [INFO] - Starting table checksum verification
was the final one, seems that it was his final word to tell on me before dying after an unsuccessful restore attempt.

**Reply by astesin** [reply]
And one more question, please. If the _Lock wait timeout exceeded; try restarting transaction_ ERROR is fatal (I mean, getting one single error of this kind means that the task of loading data resulted in a failure as a whole), shouldn't myloader abort completely and stop proceeding further with other queued jobs?

**Reply by astesin** [reply]
I've upgraded to the myloader**0.14.0-3**, built against MySQL 5.7.40-43 with SSL support with GZIP version, and will try what happens with this one and then report my observations here.
